### ISSUE TITLE
Feature request: Add a functionality for services/ports/protocols to be filtered first and then source IP

### ISSUE THREAD

**Issue by ku4eto** [firewalld/firewalld#1386]
### What would you like to be added

Feature request: Add a functionality for services/ports/protocols first and then source IP

A feature that would allow us to first filter incoming requests based on their service, port and/or protocol would be most welcome. 
This should also allow for multiple source IPs and ranges to be specified in the zones.

### Why is this needed

Currently strictly interface/source IP are the first ones to be filtered.
This is very problematic, when you have more than 1 service that you wish to allow across multiple source addresses, especially so when there are conflicting ones (allow for Service X, do not allow for Service Y). This would lead to the need to create a separate zone/rich rule for _EVERY_ single source IP.
Which kind of defeats the purpose of FirewallD being easily manageable, by having tens/hundreds/thousands of zones or rich rules.

Example:
We have a sevrer with 3 services - DNS, NTP and SSH running.
We want to allow IPs 1 to 10 (thats example) to the DNS service.
We want to allow IPs 5 to 15 to the NTP service.
We want to allow IPs 1,5,10,15,20 to the SSH service.

Easiest solution would be having 3 zones, 1 for each service with the source IPs described in each of them - which in fact we cant, since 1 source IP address or range can be defined ONLY once across all zones combined. Even if we could, if a zone matches the source IP, none of the other zones having the same Source IP will be traversed.

We are forced to either create multiple zones or a large amount of rich rules. Neither of the 2 are sane and healthy variants for managing large amounts of source IPs.

The example from above according to me would need at least 5 zones (maybe around 7).

If you increase the Source IPs, depending on the amount of overlapping IPs per service/port/protocol, the needed zones/rich rules skyrocket quite fast.

Current Zone (policy) priority cannot be used as a means of escaping the huge amount of source zones/rich rules from the example scenario, due to the same constriction of having only 1 Source IP being specified across all zones.

But Zone (policy) priority can be required to determine which service zone to be ingressed first (unless its in lexical order).

If there is similar solution, that may be implemented to allow similar scenarios to be a lot less painful experience with FirewallD, i would be fine with it as well.
Although i have the feeling the community may prefer this (due to the entire epic that was around AllowZoneDrifting).

In a previous issue i raised, `ipsets` was suggested as a solution to the given problem.
We tried it extensively and `ipsets` face the same issues with overlapping IP addresses or ranges.
Regardless if `AllowZoneDrifting` is enabled, once a rule(zone) has been processed, all subsequent rules(zones) are ignored, even if they have different IP set attached to them. This is due to the already mentioned way of work - source based rules.

**Comment by erig0** [comment]
Please don't create an identical issue. This is a duplicate of #1327.

**Comment by ku4eto** [comment]
@erig0  considering there is no way to reopen the issue, what other options do i have?

**Comment by erig0** [comment]
> considering there is no way to reopen the issue, what other options do i have?

You can continue the conversation in #1327. At present, I don't see justification to keep it open. What you're asking for is [disallowed by design](https://firewalld.org/2020/01/allowzonedrifting).
